Neos.EventSourcedNeosAdjustments/Classes/EventSourcedRouting/Projection/DocumentUriPathFinder.php

```diff
@@ -0,0 +1,146 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\EventSourcedNeosAdjustments\EventSourcedRouting\Projection;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\DBALException;
+use Neos\ContentRepository\Domain\ContentStream\ContentStreamIdentifier;
+use Neos\ContentRepository\Domain\NodeAggregate\NodeAggregateIdentifier;
+use Neos\ContentRepository\Domain\NodeAggregate\NodeName;
+use Neos\EventSourcedNeosAdjustments\EventSourcedRouting\ValueObject\DocumentNodeInfo;
+use Neos\Flow\Annotations as Flow;
+use Neos\Neos\Controller\Exception\NodeNotFoundException;
+
+/**
+ * @Flow\Scope("singleton")
+ */
+final class DocumentUriPathFinder
+{
+    private Connection $dbal;
+
+    private ?ContentStreamIdentifier $liveContentStreamIdentifierRuntimeCache = null;
+
+    public function __construct(Connection $dbal)
+    {
+        $this->dbal = $dbal;
+    }
+
+    /**
+     * @param NodeName $siteNodeName
+     * @param string $uriPath
+     * @param string $dimensionSpacePointHash
+     * @return DocumentNodeInfo
+     * @throws NodeNotFoundException
+     */
+    public function getEnabledBySiteNodeNameUriPathAndDimensionSpacePointHash(NodeName $siteNodeName, string $uriPath, string $dimensionSpacePointHash): DocumentNodeInfo
```

Comment:
I'd rename the "get..." functions here to "fetch" or "find" functions, to signify that they have a cost (with "get" functions I often assume that they might be cheaper than with fetch or find)

-> in all methods of this class

Reply:
as discussed: I named the methods `get*` by intention to emphasize that a result is expected (i.e. Exception instead of nullable return type)